Unit selection (w/ a laser pointer based impl in VR)

SeijiEmery opened the pull request:
Implemented unit selection: no movement, but movement should be able to be added trivially by calling methods on UnitSelectionManager.

Reference selection impl in VRRaycastSelector.

TBD:

- non-raycast (nearest unit within trigger sphere @ controller) selection method
- mouse selection

Additive / subtractive selection has also been implemented, though I've been unable to get the controller bindings to work in VR. I have implemented a fallback, where you can hold keyboard buttons to add or subtract from the selection. Currently (in VRSelectionTest) these are mapped to Fire1 and Fire2, ie. left ctrl (or lmb) and left alt (or rmb) respectively. 

Obviously for non-testing purposes you would not want to set these.

Note that additive / subtractive selection is not a feature that we _strictly_ need, but it could prove useful if we need it. See VRRaycastSelector.cs for a good example of how to implement this.

SeijiEmery committed:
implemented selection highlighting

SeijiEmery committed:
implemented selection

SeijiEmery committed:
Implemented additive + subtractive selection

kodomazer commented on Assets/Script/PlayerControls/UnitSelectionManager.cs:
```diff
@@ -6,15 +6,54 @@
+        {
+            unit.SetMovementDirection(normalizedMoveDir);
+        }
+    }
```
Possibly change these to:
MoveToPosition
TargetToPosition
MoveTowards
TargetTowards

To keep naming consistent between movement and targeting and allow for the possibility of incremental targeting changes as well.

SeijiEmery replied:
Yes, that would make sense - can you rename these?

kodomazer commented on Assets/Script/KelplerUnits/SelectableUnit.cs:
```diff
@@ -6,12 +6,28 @@ public class SelectableUnit : MonoBehaviour {
     // set by UnitSelectionManager 
-    public bool selected { get; set; }
+    //public bool selected { get; set; }
+    public bool selected = false;
```
Was this for editor access? Why are we leaving commented code instead of simply replacing?

SeijiEmery replied:
Yep, should’ve deleted this.

kodomazer commented on Assets/Script/KelplerUnits/UnitMovement.cs:
```diff
@@ -24,14 +24,14 @@ public class UnitMovement : MonoBehaviour {
+    public void SetFiringTarget(Vector3 targetPos) { }
 
     // sets movement direction for direct (WASD) movement
     // moveDir() should be normalized before using
```
Possibly we don't care if it's normalized before and just always normalize in the function always.

SeijiEmery replied:
Yes, that’s how this is supposed to work. Left as a note that it’s expected to be normalized though so it doesn’t do something that you didn’t expect

kodomazer reviewed (commented):
Everything looks pretty good, just a couple comments/discussion points.

brianwzhao committed:
Rename functions for movement/targeting

Renamed movement and targeting functions to be consistent
Removed some extraneous comments

kodomazer commented:
It feels weird making a commit for my own changes, but here they are.

CCD729 reviewed (approved):
In Assets/Script/PlayerControls/VRRaycastSelector.cs
Could use some explaining comment, but all looks good so far.
